core/src/style/style.h

```diff
@@ -99,10 +103,14 @@ class Style {
     /* Toggle on read if true, checks whether the context has been lost on last frame */
     bool glContextLost();
 
+    void setupShaderUniforms(int _lastBoundTex, bool _ctxLost);
+
 private:
 
     /* Whether the context has been lost on last frame */
     bool m_contextLost;
+    std::vector<StyleUniform> m_styleUniforms;
+    std::unordered_map<std::string, std::shared_ptr<Texture>> m_uniformTextures;
```

Comment:
I would prefer to have a 'textures' container in Scene and refer to  them via TextureIDs if possible. similar to FunctionID for js functions. 

In general I thought it would be good to use this approach for all Scene resources, also forStyleParams: This would also enable to have no StyleParam copying during feature style evaluation, just a pair of <StyleParamKey, StyleValueId>.